students/junyoung/westagram/users/views.py

```diff
@@ -38,5 +38,23 @@ def post(self,request):
             )
             return JsonResponse({"message" : "SUCCESS"}, status = 201)   
         
+        except KeyError as e:
+            return JsonResponse({"message" : f"KEY_ERROR : ENTER_YOUR_{e.args[0].upper()}"}, status = 400)
+
+class SignInView(View):
+    def post(self,request):
+        try:
+            data               = json.loads(request.body)
+            email              = data['email']
+            password           = data['password']
+        
+            if not User.objects.filter(email = email).exists():
+                return JsonResponse({"message" : "INVALID_USER"}, status = 401)
+        
+            elif not User.objects.filter(email = email, password = password).exists():
+                return JsonResponse({"message" : "INVALID_PASSWORD"}, status = 401)
```

Comment:
line 51,52 코드와 54, 55 코드가 둘 다 필요한가요?
```suggestion
            if not User.objects.filter(email = email).exists():
                return JsonResponse({"message" : "INVALID_USER"}, status = 401)
        
            elif not User.objects.filter(email = email, password = password).exists():
                return JsonResponse({"message" : "INVALID_PASSWORD"}, status = 401)
```

Reply:
Assignment에 계정,비밀번호 에러코드가 전부 invalid_user로 동일한 것을 못봤습니다.
54~55는 계정은 일치하는데 비밀번호가 틀리면 invalid_password를 출력하는 코드입니다.
과제에 맞게 수정하겠습니다.